### PR TITLE
LSR-954 Allow displaying Asset Publisher with porletItemName attribute in AssetDisplayScreenlet

### PR DESCRIPTION
**Pull request by dgarciasarai**

### CODE REVIEW

**Comment by dgarciasarai** [comment]
WIP: Swift 3

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java [hunk 1]
```diff
@@ -208,7 +211,7 @@ protected void onScreenletAttached() {
 	//TODO now the autoload is required to be able to load child screenlets
 	protected void autoLoad() {
 		if (SessionContext.isLoggedIn() && (entryId != 0 || (className != null
-			&& classPK != 0))) {
+			&& classPK != 0)) || (portletItemName != null && !portletItemName.equals(""))) {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java]: do the comparison the other way around

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/interactor/AssetDisplayInteractor.java [hunk 2]
```diff
@@ -72,7 +90,11 @@ protected String getIdFromArgs(Object... args) {
 		if (args.length > 1) {
 			cacheId = (long) args[1];
 		} else {
-			cacheId = (long) args[0];
+			if (args[0] instanceof Long) {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/asset/display/interactor/AssetDisplayInteractor.java]: we should move this ugly casting logic somewhere (I know there is something in the screenlet also)

**Comment by nhpatt** [comment]
LGTM after those 3 small changes 👍

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java (hunk 1 above)
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java]: Done!

**Comment by victorg1991** [comment]
LGTM 👍

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java [hunk 3]
```diff
@@ -210,8 +210,10 @@ protected void onScreenletAttached() {
 	 */
 	//TODO now the autoload is required to be able to load child screenlets
 	protected void autoLoad() {
-		if (SessionContext.isLoggedIn() && (entryId != 0 || (className != null
-			&& classPK != 0)) || (portletItemName != null && !portletItemName.equals(""))) {
+		boolean hasClassName = className != null && classPK != 0;
+		boolean hasPortletItemName = !"".equals(portletItemName);
```
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java]: Ok, understood.

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java [hunk 4]
```diff
@@ -216,8 +222,10 @@ protected void autoLoad() {
 	protected void onUserAction(String userActionName, AssetDisplayInteractor interactor, Object... args) {
 		if (entryId != 0) {
 			interactor.start(entryId);
-		} else {
+		} else if (className != null && !className.equals("") && classPK != 0) {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java]: invert plz

File: android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/MainActivity.java [hunk 5]
```diff
@@ -144,10 +145,15 @@ public void onClick(View v) {
...
+			case R.id.asset_display:
+				Intent intent1 = getIntentWithTheme(AssetDisplayActivity.class);
+				intent1.putExtra("portletItemName", getResources().getString(R.string.liferay_portlet_item_name));
+				DefaultAnimation.startActivityWithAnimation(this, intent1);
+				break;
 			case R.id.user_display:
-				Intent intent = getIntentWithTheme(AssetDisplayActivity.class);
-				intent.putExtra("entryId", Long.valueOf(getResources().getString(R.string.liferay_user_entryId)));
-				DefaultAnimation.startActivityWithAnimation(this, intent);
+				Intent intent2 = getIntentWithTheme(AssetDisplayActivity.class);
```
Comment (nhpatt) [inline android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/MainActivity.java]: better naming

**Comment by victorg1991** [comment]
👍

**Pull request merged**